app/scripts/services/validateRegistrant.js

```diff
@@ -26,9 +25,9 @@ angular.module('confRegistrationWebApp')
           }
         }
         if ((ruleOperand === 'OR' && validRuleCount > 0) || (ruleOperand === 'AND' && validRuleCount <= i)) {
-          break;
+          return false; // Exit lodash foreach as we found a case which determines the whole outcome of this function
         }
-      }
+      });
 
       return !block.rules || block.rules.length === 0 || // If no rules are set
```

Comment:
@OzzieOrca 

With the conditionals above, I would think that
```
      return !block.rules || block.rules.length === 0 || // If no rules are set
        (ruleOperand === 'OR' && validRuleCount > 0) ||
        (ruleOperand === 'AND' && validRuleCount === block.rules.length);
```

could be reduced to
```
      return !block.rules || block.rules.length === 0; // If no rules are set;
```